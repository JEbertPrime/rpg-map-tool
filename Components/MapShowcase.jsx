
import MapThumb from './MapThumb.jsx'
import styled from 'styled-components'
const Wrapper = styled.div`
    display: flex
`
export default function MapShowcase ({title, maps, count, ...props}){
    
    return(
        <div>
        {props.children}
        <Wrapper>
            
            {maps ? maps.map((map) => <MapThumb map={map} key={map._id} id={map._id} height={200} width={200} />): <div/>}
        </Wrapper>
        </div>
    )
}