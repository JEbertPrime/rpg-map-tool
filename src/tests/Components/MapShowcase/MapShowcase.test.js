import { render, screen } from "@testing-library/react";
import MapShowcase from "../../../../Components/MapShowcase";
import renderer from 'react-test-renderer';

const devMaps = [{title: 'test_thumb', fileName: 'maps/example_1.jpg', _id:1}, {title: 'test_thumb', fileName: 'maps/example_2.jpg', _id:2}]

describe("MapShowcase", () => {
  it("renders without crashing", () => {
    render(<MapShowcase maps={devMaps}><h1>test</h1></MapShowcase>);
    expect(
      screen.getByRole("heading", {name: 'test'})
    ).toBeInTheDocument();
    expect(
      screen.getAllByText('test_thumb') .length
    ).toBe(devMaps.length)

  });
  it("renders without maps correctly", ()=>{
      render(<MapShowcase >empty</MapShowcase>)
      expect(
          screen.getByText('empty')
      ).toBeInTheDocument();
  })
  it("renders correctly", ()=>{
    const tree = renderer.create(<MapShowcase maps={devMaps}><h1>test</h1></MapShowcase>).toJSON();
    expect(tree).toMatchSnapshot();

  })
});