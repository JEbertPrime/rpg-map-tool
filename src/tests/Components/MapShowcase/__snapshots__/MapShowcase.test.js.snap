// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapShowcase renders correctly 1`] = `
<div>
  <h1>
    test
  </h1>
  <div
    className="MapShowcase__Wrapper-bbss5s-0 bSIsoI"
  >
    <div
      className="MapThumb__Wrapper-sc-4r9j8g-2 hpzgfY"
    >
      <h3
        className="MapThumb__MapTitle-sc-4r9j8g-3 fHyouP"
      >
        test_thumb
      </h3>
      <div
        className="MapThumb__Parent-sc-4r9j8g-1 bUNUyg"
        id={1}
        style={
          Object {
            "backgroundImage": "url(maps/example_1.jpg)",
            "backgroundSize": "cover",
            "height": "200px",
            "width": "200px",
          }
        }
      >
        <div
          className="MapThumb__Child-sc-4r9j8g-0 dOGsgr"
        />
      </div>
    </div>
    <div
      className="MapThumb__Wrapper-sc-4r9j8g-2 hpzgfY"
    >
      <h3
        className="MapThumb__MapTitle-sc-4r9j8g-3 fHyouP"
      >
        test_thumb
      </h3>
      <div
        className="MapThumb__Parent-sc-4r9j8g-1 bUNUyg"
        id={2}
        style={
          Object {
            "backgroundImage": "url(maps/example_2.jpg)",
            "backgroundSize": "cover",
            "height": "200px",
            "width": "200px",
          }
        }
      >
        <div
          className="MapThumb__Child-sc-4r9j8g-0 dOGsgr"
        />
      </div>
    </div>
  </div>
</div>
`;
